python-threatexchange/threatexchange/cli/match_cmd.py

```diff
@@ -167,19 +167,22 @@ def execute(self, settings: CLISettings) -> None:
         if self.as_hashes:
             types = (BytesHasher, TextHasher, FileHasher)
         signal_types = [s for s in signal_types if issubclass(s, types)]
-        if self.as_hashes and len(signal_types) > 1:
-            raise CommandError(
-                f"Error: '{self.content_type.get_name()}' supports more than one SignalType."
-                " for '--hashes' also use '--only-signal' to specify one of "
-                f"{[s.get_name() for s in signal_types]}",
-                2,
-            )
-
         logging.info(
             "Signal types that apply: %s",
             ", ".join(s.get_name() for s in signal_types) or "None!",
         )
 
+        if self.as_hashes:
+            hashes_grouped_by_prefix: t.Dict[t.Optional[str], t.Set[str]] = {}
+            # Infer the signal types from the prefixes (None is used as key for hashes with no prefix)
+            for path in self.files:
+                _group_hashes_by_prefix(path, settings, hashes_grouped_by_prefix)
+            # Validate the SignalType and append the None prefixes to the correct SignalType
+            hashes_grouped_by_signal = self.validate(
+                settings, hashes_grouped_by_prefix, signal_types
+            )
+            signal_types = list(hashes_grouped_by_signal)
```

Comment:
blocking: danger! I think this will allow you jump the tracks of the signal type validation done already above. Unless this was intentional, since I actually can't think of strong benefits of narrowing it when the user clearly is passing you hashes and we enforce uniqueness on all the names. However, this will come as a surprise to the reader of the code.

```
cat pdq faceface... | tx match video -  # Surprising?
```

^ Is the above an intentional behavior change?